src/system.ts

```diff
@@ -428,12 +444,28 @@ export const chunkSource = new FullChunkSource(
   chunkDataSource,
 );
 
-const txChunksDataSource = new TxChunksDataSource({
+// Create the base TX chunks data source
+const baseTxChunksDataSource = new TxChunksDataSource({
   log,
   chainSource: arweaveClient,
   chunkSource,
 });
 
+// Create ANS-104 offset source for parsing bundle headers
+const ans104OffsetSource = new Ans104OffsetSource({
+  log,
+  dataSource: baseTxChunksDataSource,
+});
+
+// Wrap the TX chunks data source with RootParentDataSource
+// This enables resolution of data items within ANS-104 bundles
+const txChunksDataSource: ContiguousDataSource = new RootParentDataSource({
```

Comment:
Need to make this optional initially (we don't want to kill Goldsky with queries)